artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverter.java

```diff
@@ -590,9 +590,9 @@ private static ActiveMQMessage toAMQMessage(MessageReference reference,
       }
       amqMsg.setCommandId(commandId);
 
-      final SimpleString corrId = getObjectProperty(coreMessage, SimpleString.class, OpenWireConstants.JMS_CORRELATION_ID_PROPERTY);
-      if (corrId != null) {
-         amqMsg.setCorrelationId(corrId.toString());
+      final Object correlationID = coreMessage.getCorrelationID();
+      if (correlationID != null) {
+         amqMsg.setCorrelationId(correlationID.toString());
```

Comment:
The old code looks like it would get null if the types didnt match, and so not actually add thing...this looks more like it will start adding the toString() of non-Strings. Perhaps it should still be type-checking to avoid that, given the other changes make it more likely for this to receive non-String values now?

Reply:
(Or also, as Tim suggested, potentially try UTF8 encoding e.g bytes received at this point).

Reply:
I think the actual-expected-behaviour should be tested. This specific behaviour wasnt noticed as there is a simply a 'disabled test as it wont send matching bytes', which is not covering what actually was/is expected to happen in this situation either before or after the changes here.

Reply:
I added a couple tests which I believe should cover the relevant possibilities.

Reply:
So the expectation is an Openwire client will now always receive null if a non-Openwire sender sends byte[], as opposed to previously getting a string-encoding of the value if the sender was AMQP?

Seems like it we are going to change it, it could instead try sending the bytes as a string, if they are UTF-8...or keep doing what it did before for AMQP.

Reply:
@gemmellr, is this ready to merge in your opinion?

Reply:
I commented on a new aspect related to changes made from this comment, lets move over there :)